### PR TITLE
api: Add support for verbose inventory quantities

### PR DESCRIPTION
**Pull request by ImNoOSRS**
A toggle which makes it possible to show full lenght inventory quantities.

### CODE REVIEW

**Comment by Smeety** [comment]
;D

File: runelite-api/src/main/java/net/runelite/api/Client.java [hunk 1]
```diff
@@ -1704,6 +1704,13 @@
 	 */
 	void setCompass(Sprite Sprite);
 
+	/**
+	 * Sets whether inventory quantity is verbose.
+	 *
+	 * @param state verbose state
+	 */
+	void setVerbose(boolean state);
```
Comment (Lucwousin) [inline runelite-api/src/main/java/net/runelite/api/Client.java:1712]: Change this method name to actually describe what's made verbose (ie setItemQuantitiesVerbose)

File: runelite-mixins/src/main/java/net/runelite/mixins/VerboseInventoryMixin.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package net.runelite.mixins;
```
Comment (Lucwousin) [inline runelite-mixins/src/main/java/net/runelite/mixins/VerboseInventoryMixin.java:1]: This is missing a copyright header

File: runelite-mixins/src/main/java/net/runelite/mixins/VerboseInventoryMixin.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import java.awt.Dimension;
+import net.runelite.api.Constants;
+import net.runelite.api.mixins.Copy;
+import net.runelite.api.mixins.Inject;
+import net.runelite.api.mixins.Mixin;
+import net.runelite.api.mixins.Replace;
+import net.runelite.rs.api.RSClient;
+
+@Mixin(RSClient.class)
+public abstract class VerboseInventoryMixin implements RSClient
```
Comment (Lucwousin) [inline runelite-mixins/src/main/java/net/runelite/mixins/VerboseInventoryMixin.java:13]: VerboseInventoryQuantitiesMixin? It only makes the quantities more verbose, not the inventory in it's entirety

File: runelite-mixins/src/main/java/net/runelite/mixins/VerboseInventoryMixin.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+{
+    @Inject
+    private static boolean verbose;
+
+    @Inject
+    public void setVerbose(boolean state)
```
Comment (Lucwousin) [inline runelite-mixins/src/main/java/net/runelite/mixins/VerboseInventoryMixin.java:19]: Add `@Override` to this, so you can tell it's implementation for the interface more easily

File: runelite-mixins/src/main/java/net/runelite/mixins/VerboseInventoryMixin.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    @Copy("inventoryQuantityFormat")
+    @Replace("inventoryQuantityFormat")
+    public static String getInventoryQuantityFormat(int n) {
+        if (verbose) {
+            return String.valueOf(n);
+        }
+
+        long decimal = n;
```
Comment (Lucwousin) [inline runelite-mixins/src/main/java/net/runelite/mixins/VerboseInventoryMixin.java:31]: Remove this unused variable

**Review by Lucwousin — CHANGES_REQUESTED** [review]
You should also change the commit message, `-` isn't really descriptive. Other than that, looks good! Make sure to also create a plugin which utilizes this functionality :P

File: runescape-client/src/main/java/class194.java [hunk 6]
```diff
@@ -79,7 +79,8 @@
 		descriptor = "(IS)Ljava/lang/String;",
 		garbageValue = "4096"
 	)
-	static final String method3712(int var0) {
+	@Export("inventoryQuantityFormat")
```
Comment (Lucwousin) [inline runescape-client/src/main/java/class194.java:82]: there's no import for Export

**Comment by ImNoOSRS** [comment]
Thanks for the edits @Lucwousin i really do pull requests so its quite new to me i tried renaming the commit from - to something else but i didnt really figure out how to do that

**Review by Lucwousin — APPROVED** [review]

**Pull request merged**